api/controllers/PromotionController.js

```diff
@@ -41,17 +41,24 @@ var PromotionController = {
 
     getRestaurantById: function(req, res) {
         var id = req.swagger.params.id.value;
+        PromotionController.getRestaurantAndDoSomething(id, PromotionController.returnRestaurantDetail, res, null);
+    },
 
+    getRestaurantAndDoSomething: function(id, callback, res, params) {
         superagent
         .get('https://api.yelp.com/v3/businesses/'+ id)
         .set('Authorization', yelpAPIKEY)
         .set('Accept', 'application/json')
         .end((err, data) => {
             if (err) { return console.log(err); }
 
-            return res.json({restaurant : data.body});
+            callback(data, res, params);
         });
     },
+
+    returnRestaurantDetail: function(data, res){
+        return res.json({restaurant : data.body});
```

Comment:
not sure it's necessary to have a one-line body method

Reply:
I couldn't think of another way of returning the final reault.
This funciton is solely called for returning our restaurant detail.

Reply:
Any ideas?